src/plugins/plugin-mapbox-gl/index.js

```diff
@@ -0,0 +1,235 @@
+
+import sortBy from 'lodash/sortBy';
+
+import rasterLayer from './raster-layer-mapbox-gl';
+import vectorLayer from './vector-layer-mapbox-gl';
+import geoJsonLayer from './geojson-layer-mapbox-gl';
+
+class PluginMapboxGL {
+  constructor(map, options) {
+    this.map = map;
+    this.options = options;
+
+    // You can change mapStyles and all the layers will be repositioned
+    this.map.on('style.load', () => {
+      const { getLayers } = this.options;
+      const layers = getLayers();
+
+      layers.forEach(layer => this.add(layer, layers));
+    });
+  }
+
+  method = {
+    leaflet: rasterLayer,
```

Comment:
This is to prevent a mass update in the API config for layers and also to prevent breaking changes. Ideally once we have this stable and we are happy we can remove this and release a v2.

Reply:
You're already using it on prod. So why not dissociate what you're using in prod from what lands into layer-manager. Breaking changes can happen, in this case I would see as an option adding a new type there's no need to call it leaflet.